insights/client/utilities.py

```diff
@@ -318,6 +311,33 @@ def systemd_notify(pid):
         logger.debug('systemd-notify returned %s', proc.returncode)
 
 
+def systemd_notify_init_thread():
+    '''
+    Use a thread to periodically ping systemd instead
+    of calling it on a per-command basis
+    '''
+    pid = read_pidfile()
+    if not pid:
+        logger.debug('No PID specified.')
+        return
+    if not os.getenv('NOTIFY_SOCKET'):
+        # running standalone, not via systemd job
+        return
+    if not os.path.exists('/usr/bin/systemd-notify'):
+        # RHEL 6, no systemd
+        return
+
+    def _sdnotify_loop():
+        while True:
+            # run sdnotify every 30 seconds
+            _systemd_notify(pid)
```

Comment:
Would it make sense to hook up an exit condition to this thread function to the return value of `_systemd_notify`? Something like:

```python
def _sdnotify_loop():
    ret = True
    while ret:
        global ret  # I think this is how Python's goofy variable scoping works...
        ret = _systemd_notify(pid)
```

That way if `_systemd_notify` needs to abort for any reason (like a non-zero exit code from `systemd-notify`, or a Popen failure trying to create the child process), it can safely abort the thread.